tests/pytests/unit/grains/test_core.py

```diff
@@ -3593,3 +3706,254 @@ def _mock_is_file(filename):
 
         assert virtual_grains["virtual"] == "Nitro"
         assert virtual_grains["virtual_subtype"] == "Amazon EC2"
+
+
+def test_append_domain():
+    """
+    test append_domain
+    """
+
+    assert core.append_domain() == {}
+
+    with patch.object(salt.utils.platform, "is_proxy", MagicMock(return_value=True)):
+        assert core.append_domain() == {}
+
+    with patch("salt.grains.core.__opts__", {"append_domain": "example.com"}):
+        assert core.append_domain() == {"append_domain": "example.com"}
+
+
+def test_hostname():
+    """
+    test append_domain
+    """
+
+    with patch.object(salt.utils.platform, "is_proxy", MagicMock(return_value=True)):
+        assert core.hostname() == {}
+
+    with patch("salt.grains.core.__FQDN__", None), patch(
+        "socket.gethostname", MagicMock(return_value=None)
+    ), patch("salt.utils.network.get_fqhostname", MagicMock(return_value=None)):
+        assert core.hostname() == {
+            "localhost": None,
+            "fqdn": "localhost.localdomain",
+            "host": "localhost",
+            "domain": "localdomain",
+        }
+
+
+def test_zmqversion():
+    """
+    test zmqversion
+    """
+
+    ret = core.zmqversion()
+    assert "zmqversion" in ret
+
+    with patch.dict("sys.modules", {"zmq": None}):
+        ret = core.zmqversion()
+        assert "zmqversion" not in ret
+
+
+def test_saltpath():
+    """
+    test saltpath
+    """
+
+    ret = core.saltpath()
+    assert "saltpath" in ret
+
+
+def test_pythonexecutable():
+    """
+    test pythonexecutable
+    """
+
+    ret = core.pythonexecutable()
+    assert "pythonexecutable" in ret
```

Comment:
Can we test for not just the key but also the value in the dictionary that is added. This comment also relates to `test_pythonpath` and `test_pythonversion`